service/external_paysession.go

```diff
@@ -0,0 +1,46 @@
+package service
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/companieshouse/payments.api.ch.gov.uk/models"
+
+	"github.com/companieshouse/chs.go/log"
+)
+
+// CreateExternalPaymentJourney creates an external payment session with a Payment Provider that is given, e.g: GovPay
+func CreateExternalPaymentJourney(w http.ResponseWriter, req *http.Request) {
+	if req.Body == nil {
+		log.ErrorR(req, fmt.Errorf("request body empty"))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	requestDecoder := json.NewDecoder(req.Body)
+	var incomingExternalPaymentJourneyRequest models.IncomingExternalPaymentJourneyRequest
+	err := requestDecoder.Decode(&incomingExternalPaymentJourneyRequest)
```

Comment:
coud probably just put this on the next line down?
```go
if requestDecoder.Decode(&incomingExternalPaymentJourneyRequest != nil {
    log.ErrorR(req, fmt.Errorf("request body invalid: [%v]", err))
    w.WriteHeader(http.StatusBadRequest)
    return
}
```

Reply:
@ChrisJBurns That would prevent `err` being declared to add to the log.

Reply:
Such a stupid mistake by me :P ... ignore me